products/compute/api.yaml

```diff
@@ -11452,32 +11527,742 @@ objects:
+                          name: 'queryParameterMatches'
+                          description: |
+                            Specifies a list of query parameter match criteria, all of which must match
+                            corresponding query parameters in the request.
+                          item_type: !ruby/object:Api::Type::NestedObject
+                            properties:
+                              - !ruby/object:Api::Type::String
+                                name: 'exactMatch'
+                                description: |
+                                  The queryParameterMatch matches if the value of the parameter exactly matches
+                                  the contents of exactMatch. Only one of presentMatch, exactMatch and regexMatch
+                                  must be set.
+                              - !ruby/object:Api::Type::String
+                                name: 'name'
+                                description: |
+                                  The name of the query parameter to match. The query parameter must exist in the
+                                  request, in the absence of which the request match fails.
+                              - !ruby/object:Api::Type::Boolean
+                                name: 'presentMatch'
+                                description: |
+                                  Specifies that the queryParameterMatch matches if the request contains the query
+                                  parameter, irrespective of whether the parameter has a value or not. Only one of
+                                  presentMatch, exactMatch and regexMatch must be set.
+                              - !ruby/object:Api::Type::String
+                                name: 'regexMatch'
+                                description: |
+                                  The queryParameterMatch matches if the value of the parameter matches the
+                                  regular expression specified by regexMatch. For the regular expression grammar,
+                                  please see en.cppreference.com/w/cpp/regex/ecmascript  Only one of presentMatch,
+                                  exactMatch and regexMatch must be set.
+                        - !ruby/object:Api::Type::String
+                          name: 'regexMatch'
+                          description: |
+                            For satifying the matchRule condition, the path of the request must satisfy the
+                            regular expression specified in regexMatch after removing any query parameters
+                            and anchor supplied with the original URL. For regular expression grammar please
+                            see en.cppreference.com/w/cpp/regex/ecmascript  Only one of prefixMatch,
+                            fullPathMatch or regexMatch must be specified.
+                  - !ruby/object:Api::Type::NestedObject
```

Comment:
> Only one of routeAction or urlRedirect must be set.

Can we use `ExactlyOneOf`?

Reply:
I don't think so because there can be more than one element in `routeRules`